exploration/number-selection.md

```diff
@@ -53,6 +53,21 @@ Both JS and ICU PluralRules implementations provide for determining the plural c
 of a range based on its start and end values.
 Range-based selectors are not initially considered here.
 
+In <a href="https://github.com/unicode-org/message-format-wg/pull/842">PR #842</a>
+@eemeli points out a number of gaps or infelicities in the current specification
+and there was extensive discussion of how to address these gaps.
+
+The `key` for exact numeric match in a variant has to be a string. 
```

Comment:
I don't see how this results from the requirements.

In some cases the key has to be a string, in other cases it is enough to be a number.
So the whole section below is only one option: IF we consider the keys to be stings, then ...

The idea that the key can be a number sometimes is not considered.

But it would be natural to map "...foo {}..." and "...|foo| {}..." in syntax to strings, and "...123 {}..." and "...|123| {}..." in syntax to numbers.

Reply:
The key has to be a string because the message is a string. The next line addresses this: if the key is a string, then the format of the string has to be clear so that it can be related to a number.

Reply:
> The key has to be a string because the message is a string

I don't see how that one results from the other.
What says that keys and messages should be the same type?
And even if there is something, nothing stops us from changing it.